### ISSUE TITLE
Improper handling of value with an apostrophe in valueProcessor()

### ISSUE THREAD

**Issue by lionturtle99** [react-querybuilder/react-querybuilder#485]
**Improper handling of value with an apostrophe in valueProcessor()**

I found a bug with the `valueProcessor() ` when there's a single quote/apostrophe inside a string. For example, formatting the query as SQL you'd expect `Name = "Amerigo's"` or `Name = 'Amerigo''s'` (standard ANSI SQL escaped quotes), but instead it returns `Name = 'Amerigo's'` which is invalid syntax.

An example of the issue can be seen is this [code sandbox](https://stackblitz.com/edit/github-snvkft-u1zhcu?file=src/App.tsx).

Thank you.

**Comment by jakeboone02** [comment]
This ended up being a pretty simple fix and I'll release a patch later today, but also it's something you can avoid now (v6, and I think even in v5) by using `ruleProcessor` instead of `valueProcessor`. They accept the same arguments, so you can just assign your custom function like this:

```ts
formatQuery(query, { format: 'sql', ruleProcessor: customProcessor })
```

instead of this:

```ts
formatQuery(query, { format: 'sql', valueProcessor: customProcessor })
```

You can play around with the latest, unreleases RQB build [here](https://codesandbox.io/p/sandbox/react-querybuilder-ci-forked-qqm2gr?welcome=true&file=%2FREADME.md). That sandbox should demonstrate that the issue is resolved.

An updated, working version of your example is [here](https://stackblitz.com/edit/github-snvkft-i9jsn9?file=src%2FApp.tsx).

I'll update the documentation to make it clearer that `ruleProcessor` is preferred/recommended.

**Comment by jakeboone02** [comment]
This issue was resolved in [v6.0.7](https://github.com/react-querybuilder/react-querybuilder/releases/tag/v6.0.7).
